src/VisualStudio/Core/Next/Remote/ServiceHubRemoteHostClient.cs

```diff
@@ -31,6 +39,10 @@ internal sealed partial class ServiceHubRemoteHostClient : RemoteHostClient
         private readonly JsonRpc _rpc;
         private readonly ReferenceCountedDisposable<RemotableDataJsonRpc> _remotableDataRpc;
 
+        private readonly object _globalNotificationsGate = new object();
+        private Task<GlobalNotificationState> _globalNotificationsTask = Task.FromResult(GlobalNotificationState.NotStarted);
+        private KeepAliveSession _keepAliveSession;
```

Comment:
@CyrusNajmabadi sorry about confusion. I think you don't actually need KeepAliveSession. Client already has _rpc that is connected to RemoteHostService. you can just use that one to send it over.